File: demos/iOS/Example-iOS-ObjC/Example-iOS-ObjC/PP_PeripheralListController.m
//
//  PP_PeripheralListController.m
//  Example-iOS-ObjC
//
//  Created by Hildequias Junior on 8/8/17.
//  Copyright © 2017 UOL Pagseguro. All rights reserved.
//

#import "PP_PeripheralListController.h"
#import "PlugPag.h"
#import "UIUtils.h"

#define MSG_PAIR_SUCESS       @"Pareamento efetuado com sucesso!"

@interface PP_PeripheralListController () {
    NSArray *devices;
}

@end

@implementation PP_PeripheralListController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [[PlugPag sharedInstance] SetVersionName:@"UnitTest" withVersion:@"V001"];
}

- (void) viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - Table view data source
- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return [devices count];
}


- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    
    static NSString *CellIdentifier = @"CellID";
    
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:CellIdentifier forIndexPath:indexPath];
    cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
    NSString *messageAtIndexPath = [devices objectAtIndex :[indexPath row]];
    [[cell textLabel] setText:messageAtIndexPath];
    
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    
    NSString *deviceChosen = [devices objectAtIndex:indexPath.row];
    
    [[PlugPag sharedInstance] SetPeripheralName:deviceChosen];
    
    [UIUtils showAlert:self withMessage: [NSString stringWithFormat:@"Selecionado: %@", deviceChosen]];
    
    [self.tableView deselectRowAtIndexPath:indexPath animated:YES];
}

- (IBAction)scan:(id)sender {
    
    [self scanPeripherals];
}

- (void) scanPeripherals {
    
    [UIUtils showProgress];
    
    devices = [[PlugPag sharedInstance] GetListPeripheral];
    [self.tableView reloadData];
    
    [UIUtils hideProgress];
}

@end

File: demos/iOS/Example-iOS-Swift/Example-iOS-Swift/PP_PeripheralListController.swift
//
//  PP_PeripheralListController.swift
//  Example-iOS-Swift
//
//  Created by Hildequias Junior on 8/23/17.
//  Copyright © 2017 UOL Pagseguro. All rights reserved.
//

import UIKit

class PP_PeripheralListController: UITableViewController {

    var devices: NSArray! = NSArray()
    
    // MARK: - UIViewController Cycle
    override func viewDidLoad() {
        super.viewDidLoad()
        
        PlugPag.sharedInstance().setVersionName("UnitTest", withVersion: "V001")
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    // MARK: - Table view data source
    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return devices.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        let CellIdentifier = "CellID"
        
        // Configure the cell...
        let cell = tableView.dequeueReusableCell(withIdentifier: CellIdentifier, for: indexPath)
        cell.accessoryType = .disclosureIndicator
        cell.textLabel?.text = devices.object(at: indexPath.row) as? String
        
        return cell
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        
        let deviceChosen = devices.object(at: indexPath.row) as! String
        
        PlugPag.sharedInstance().setPeripheralName(deviceChosen)
        
        UIUtils.showAlert(view: self, message: "Selecionado: \(deviceChosen)")
        
        tableView.deselectRow(at: indexPath, animated: true)
    }
    
    // MARK: - IBAction Buttons
    @IBAction func scan(sender: UIBarButtonItem) {
        
        scanPeripherals()
    }
    
    // MARK: - Business Methods
    func scanPeripherals() {
        
        UIUtils.showProgress()
        
        devices = PlugPag.sharedInstance().getListPeripheral()! as NSArray
        
        self.tableView.reloadData()
        UIUtils.hideProgress()
    }
}
